openpgp/packet/signature.go

```diff
@@ -550,6 +560,11 @@ func (sig *Signature) Sign(h hash.Hash, priv *PrivateKey, config *Config) (err e
 		return
 	}
 
+	if hashIsSigner {
+		err = signer.Sign(sig)
```

Comment:
...and this one here, though, not having followed it all the way through, I understand what you're saying that these might not be new error cases...